proposals/lip-0045.md

```diff
@@ -984,164 +834,351 @@ The Interoperability module contains the following commands:
+3. `CCM_PROCESSED_STATUS_COMMAND_NOT_SUPPORTED`: The receiving chain does not support the command.
+4. `CCM_PROCESSED_STATUS_FORWARDED`: The CCM was forwarded to the receiving chain.
+5. `CCM_PROCESSED_STATUS_APPLIED`: The CCM was applied on the receiving chain.
+6. `CCM_PROCESSED_STATUS_DISCARDED`: The CCM was discarded.
+7. `CCM_PROCESSED_STATUS_RECOVERY_FORWARDED`: The CCM was forwarded to the original sending chain for recovery.
+8. `CCM_PROCESSED_STATUS_RECOVERY_APPLIED`: The CCM was recovered and applied on the original sending chain.
+9. `CCM_PROCESSED_STATUS_RECOVERY_DISCARDED`: The CCM was recovered and discarded.
 
-* `chainID`: The ID of the chain.
 
-##### Returns
+##### Topics
 
-This function returns `chainAccount(chainID)`.
+- `sendingChainID`: The ID of the sending chain.
+- `receivingChainID`: The ID of the receiving chain.
 
-#### getChannel
+##### Data
 
-The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID`.
+```java
+ccmProcessedDataSchema = {
+    "type": "object",
+    "required": ["ccmID", "status"],
+    "properties": {
+        "ccmID": {
+            "dataType": "bytes",
+            "length": HASH_LENGTH, 
+            "fieldNumber": 1
+        },
+        "status": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
+```
 
-##### Parameters
+### Protocol Logic for Other Modules
 
-* `chainID`: The ID of the chain.
+#### send
+
+The `send` function is used to create and add a message to the outbox of a partner chain.
+
+```python
+def send(sendingAddress: Address, 
+    module: Module, 
+    crossChainCommand: CrossChainCommand, 
+    receivingChainID: ChainID, 
+    fee: uint64, 
+    status: uint32, 
+    params: object
+) -> None:
+    # Processing on the mainchain
+    if ownChainAccount.chainID == CHAIN_ID_MAINCHAIN:
+        # Chain has to exist on mainchain
+        if chainAccount(receivingChainID) does not exists:
+            emitPersistentEvent(
+                module = MODULE_NAME_INTEROPERABILITY,
+                name = EVENT_NAME_CCM_SENT,
+                data = {"ccmID": ccmID, "status": CCM_PROCESSED_STATUS_CHANNEL_UNAVAILABLE},
+                topics = [ccm.sendingChainID, ccm.receivingChainID, ccmID]
+            )
+            raise Exception('Receiving chain does not exist.')
+        partnerChainID = receivingChainID
+    # Processing on a sidechain
+    else:
+        # Check for direct channel
+        if chainAccount(receivingChainID) does not exists:
+            partnerChainID = mainchainID
+        else:
+            partnerChainID = receivingChainID
 
-##### Returns
+    # Chain must be live; This checks is always on the receivingChainID
+    if not isLive(receivingChainID):
+        emitPersistentEvent(
+            module = MODULE_NAME_INTEROPERABILITY,
+            name = EVENT_NAME_CCM_SENT,
+            data = {"ccmID": ccmID, "status": CCM_SENT_STATUS_CHANNEL_UNAVAILABLE},
```

Comment:
Do you not want to distinguish the cases if the receiving chain is not live (here) or not active (below). Maybe this could allow the front-end to be more specific why sending a CCM failed.

Reply:
I think this is fine-grained enough